pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/JsonExtractKeyTransformFunction.java

```diff
@@ -40,8 +41,12 @@
  *
  * Usage:
  * jsonExtractKey(jsonFieldName, 'jsonPath')
+ * jsonExtractKey(jsonFieldName, 'jsonPath', maxDepth)
+ * jsonExtractKey(jsonFieldName, 'jsonPath', maxDepth, dotNotation)
```

Comment:
For the optional parameters, it is not easy to use if we force the order, e.g. when user only want to set `dotNotation` but not `maxDepth`
One way to handle it is to make it one single argument keyed by the parameter name. See `DistinctCountSmartHLLAggregationFunction.Parameters` as an example